numpyro/distributions/util.py

```diff
@@ -215,6 +215,20 @@ def multinomial(key, p, n, shape=()):
     return _multinomial(key, p, n, n_max, shape)
 
 
+@partial(jit, static_argnums=(2,))
+def _geometric(key, p, shape):
+    if shape is None:
+        shape = jnp.shape(p)
+    dtype = get_dtype(p)
+    tiny = jnp.finfo(dtype).eps
+    u = random.uniform(key, shape, dtype, tiny)
```

Comment:
nit: because 1 is a valid value for `u`, I guess we can take `u = random.uniform(key, shape, dtype)` and compute `jnp.log1p(-u)` instead of `jnp.log(u)` without having to clamp the value.

Reply:
I follow the implementation in torch https://github.com/pytorch/pytorch/blob/master/torch/distributions/geometric.py#L78
Also, could `log1p(-u)` be little unstable when `u -> 1`?

Reply:
In most frameworks, `uniform` samples values in [0, 1), so it should be fine. We can use the trick `1 - u` to get a uniform sample in (0, 1]. If you prefer to sample from (eps, 1), feel free to ignore my comment.

Reply:
Oh, I got that.